### PR TITLE
ci: fix excessive token permissions

### LINKED ISSUE #14466: github workflows  with excessive token-permissions
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

the OpenSSF scorecard report https://scorecard.dev/viewer/?uri=github.com/rook/rook

reports excessive token-permissions in rook's github workflows:

```

Token-Permissions
high
Determines if the project's workflows follow the principle of least privilege.
Reason
detected GitHub workflow tokens with excessive permissions
Details
Info: jobLevel 'contents' permission set to 'read': .github/workflows/commitlint.yml:25
Info: jobLevel 'pull-requests' permission set to 'read': .github/workflows/commitlint.yml:26
Warn: no topLevel permission defined: .github/workflows/auto-assign.yaml:1
Info: topLevel 'contents' permission set to 'read': .github/workflows/build.yml:16
Warn: no topLevel permission defined: .github/workflows/canary-integration-suite.yml:1
Warn: no topLevel permission defined: .github/workflows/canary-integration-test.yml:1
Info: topLevel 'contents' permission set to 'read': .github/workflows/codegen.yml:25
Info: topLevel 'contents' permission set to 'read': .github/workflows/codespell.yaml:20
Info: topLevel 'contents' permission set to 'read': .github/workflows/commitlint.yml:20
Info: topLevel 'contents' permission set to 'read': .github/workflows/crds-gen.yml:25
Warn: no topLevel permission defined: .github/workflows/create-tag.yaml:1
Warn: no topLevel permission defined: .github/workflows/daily-nightly-jobs.yml:1
Info: topLevel 'contents' permission set to 'read': .github/workflows/docs-check.yml:20
Info: topLevel 'contents' permission set to 'read': .github/workflows/golangci-lint.yaml:20
Info: topLevel 'contents' permission set to 'read': .github/workflows/helm-lint.yaml:22
Warn: no topLevel permission defined: .github/workflows/integration-test-helm-suite.yaml:1
Warn: no topLevel permission defined: .github/workflows/integration-test-mgr-suite.yaml:1
Warn: no topLevel permission defined: .github/workflows/integration-test-multi-cluster-suite.yaml:1
Warn: no topLevel permission defined: .github/workflows/integration-test-object-suite.yaml:1
Warn: no topLevel permission defined: .github/workflows/integration-test-smoke-suite.yaml:1
Warn: no topLevel permission defined: .github/workflows/integration-test-upgrade-suite.yaml:1
Warn: no topLevel permission defined: .github/workflows/integration-tests-on-release.yaml:1
Info: topLevel 'contents' permission set to 'read': .github/workflows/linters.yaml:20
Info: topLevel 'contents' permission set to 'read': .github/workflows/mod-check.yml:25
Warn: no topLevel permission defined: .github/workflows/multus.yaml:1
Info: topLevel 'contents' permission set to 'read': .github/workflows/push-build.yaml:16
Info: topLevel 'contents' permission set to 'read': .github/workflows/rbac-gen.yaml:25
Info: topLevel 'contents' permission set to 'read': .github/workflows/shellcheck.yaml:20
Info: topLevel 'contents' permission set to 'read': .github/workflows/snyk.yaml:11
Info: topLevel 'contents' permission set to 'read': .github/workflows/stale.yaml:8
Info: topLevel 'contents' permission set to 'read': .github/workflows/unit-test.yml:32
Info: no jobLevel write permissions found
```

**Expected behavior:**

Token-Permissions should score 10  on the board. 

**How to reproduce it (minimal and precise):**

N/A

**File(s) to submit**:

N/A

**Logs to submit**:
N/A

**Cluster Status to submit**:

N/A

* Output of kubectl commands, if necessary

N/A

**Environment**:

N/A

### PR DESCRIPTION
**Pull request by obnoxxx**
The OpenSSF scorecard report

https://scorecard.dev/viewer/?uri=github.com/rook/rook

warns about excessive token-permissions in the rook project's github CI workflows.

This bulk change aims at improving this by

1. adding top-level read permissions where missing for integration tests.
2. adding write permissions as needed for the auto-assign workflow.
3. removing the warned-about and unused create-tag workflow

**Issue resolved by this Pull Request:**
Resolves #14466 

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/Contributing/development-flow/#commit-structure).
- [x] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/Contributing/development-flow/#submitting-a-pull-request)
- [x] [Pending release notes](https://github.com/rook/rook/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [x] Integration tests have been added, if necessary.

### CODE REVIEW

File: .github/workflows/canary-integration-suite.yml [hunk 1]
```diff
@@ -24,6 +24,11 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
   cancel-in-progress: true
 
+  permissions:
+    contents: read
+
+
+
```
Comment (BlaineEXE) [inline .github/workflows/canary-integration-suite.yml:31]: Let's keep these files following the pattern of one empty line between sections. I see between 0 and 3 across all files.

File: .github/workflows/auto-assign.yaml [hunk 2]
```diff
@@ -5,6 +5,9 @@ on:
 
 jobs:
   assign:
+    permissions:
+      @ write permissions needed to assign the issue.
+      contents: write
```
Comment (BlaineEXE) [inline .github/workflows/auto-assign.yaml:10]: Nice to see we can limit permissions to just jobs.

I wonder if the best-practice should have us putting default read permissions on the top-level in addition to the job-necessary write?

```suggestion
permissions:
  contents: read

jobs:
  assign:
    permissions:
      @ write permissions needed to assign the issue.
      contents: write
```

File: .github/workflows/canary-integration-suite.yml (hunk 1 above)
Comment (obnoxxx) [reply .github/workflows/canary-integration-suite.yml]: @BlaineEXE  wrote:

> Let's keep these files following the pattern of one empty line between sections. I see between 0 and 3 across all files.

Good point. I tried to find and fix all cases.

File: .github/workflows/integration-test-multi-cluster-suite.yaml [hunk 3]
```diff
@@ -18,6 +18,9 @@ concurrency:
   group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
   cancel-in-progress: true
 
+  permissions:
+    contents: read
```
Comment (BlaineEXE) [inline .github/workflows/integration-test-multi-cluster-suite.yaml:22]: I'm pretty sure the top-level permissions config needs to be left aligned. As written, I think this interprets as being part of the concurrency section above.

```suggestion
permissions:
  contents: read
```

File: .github/workflows/auto-assign.yaml (hunk 2 above)
Comment (BlaineEXE) [reply .github/workflows/auto-assign.yaml]: Thoughts here @obnoxxx ?

Comment (obnoxxx) [reply .github/workflows/auto-assign.yaml]: @BlaineEXE  wrote:

> Nice to see we can limit permissions to just jobs.

Thanks for confirming the approach!

> I wonder if the best-practice should have us putting default read permissions on the top-level in addition to the job-necessary write?

great idea, thanks! done.

File: .github/workflows/integration-test-multi-cluster-suite.yaml (hunk 3 above)
Comment (BlaineEXE) [reply .github/workflows/integration-test-multi-cluster-suite.yaml]: I still see a handful of places where this is indented at the top level.

Comment (obnoxxx) [reply .github/workflows/integration-test-multi-cluster-suite.yaml]: @BlaineEXE  wrote:

> I still see a handful of places where this is indented at the top level.

I think now I have found and fixed all cases

**Review by BlaineEXE — APPROVED** [review]

File: .github/workflows/auto-assign.yaml [hunk 4]
```diff
@@ -3,8 +3,14 @@ on:
   issue_comment:
     types: [created, edited]
 
+permissions:
+  contents: read
+
 jobs:
   assign:
+    permissions:
+      # write permissions areneeded to assign the issue.
```
Comment (subhamkrai) [inline .github/workflows/auto-assign.yaml:12]:
```suggestion
      # write permissions are needed to assign the issue.
```
?

**Review by subhamkrai — CHANGES_REQUESTED** [review]

**Review by subhamkrai — COMMENTED** [review]
@obnoxxx you can remove my name from `co-authored-by` 😄

File: .github/workflows/auto-assign.yaml [hunk 5]
```diff
@@ -3,8 +3,14 @@ on:
...
     types: [created, edited]
 
+permissions:
+  contents: read
+
 jobs:
   assign:
+    permissions:
+      # write permissions are needed to assign the issue.
+      contents: write
```
Comment (parth-gr) [inline .github/workflows/auto-assign.yaml:13]: can you try this once this get merged

**Comment by obnoxxx** [comment]
@subhamkrai  wrote:
> @obnoxxx you can remove my name from `co-authored-by` 😄

This came from accepting your suggestion and I usually like to give credit, but should have put your real email address.

Anyway: removed as requested. :)

**Comment by subhamkrai** [comment]
> @subhamkrai wrote:
> 
> > @obnoxxx you can remove my name from `co-authored-by` 😄
> 
> This came from accepting your suggestion and I usually like to give credit, but should have put your real email address.
> 
> Anyway: removed as requested. :)

thank you

**Review by subhamkrai — APPROVED** [review]

**Pull request merged**